app/modules/wifi.c

```diff
@@ -1969,28 +1969,24 @@ void wifi_change_default_host_name(void)
   uint8 opmode_temp=wifi_get_opmode();
   wifi_set_opmode_current(STATION_MODE);
   char temp[33] = {0};//32 chars + NULL
+
+  const char *hostname = "NODE";
+#ifdef WIFI_STA_HOSTNAME
+  if (wifi_sta_checkhostname(WIFI_STA_HOSTNAME, strlen(WIFI_STA_HOSTNAME))) {
```

Comment:
If wifi_sta_checkhostname fails and !defined(WIFI_STA_HOSTNAME_APPEND_MAC)
The name will be just NODE

Reply:
On reflection, I think that the possibly better approach is to handle an invalid hostname in the WIFI_STA_HOSTNAME by ignoring this entirely and reverting to the espressif default. It would simplify the code some more, and I really don't want to have this weird fallback handling.....

Reply:
Agreed. 
Apart from that it would be nice if it would be possible to show the failure of
wifi_sta_checkhostname to the user
Maybe just use a name like  "invalid Hostname" then.  Or mention the possibility of invalid Hostname should be described where it is defined

Reply:
I've added this